src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,43 @@
+.calendar {
+  width: 7 * ($day-size + $gap) - $gap + 2 * $calendar-padding;
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap;
+  padding: $calendar-padding;
+
+  &__day {
+    height: $day-size;
+    width: $day-size;
+    border: 1px solid #000;
+    background-color: #eee;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    transition-duration: 0.5s;
+
+    @for $i from 1 through $default-month-length {
+      &:nth-child(#{$i})::before {
+        content: '#{$i}';
+      }
+    }
+
+    &:hover {
+      cursor: pointer;
+      background-color: #ffbfcb;
+      transform: translateY(-20px);
+      transition-duration: 0.5s;
+    }
+  }
+
+  @each $day, $day-number in $days-of-week {
+    &--start-day-#{$day} :first-child {
+      margin-left: (($day-size + $gap) * $day-number);
```

Comment:
Ensure that the calculated 'margin-left' does not result in a value that exceeds the container's width, which could cause layout issues. It's good practice to add a check to ensure that the 'margin-left' applied to the ':first-child' does not push the element outside the visible area of the '.calendar' block.